File: C#/BluffinMuffin.Protocol/Lobby/RegisteredMode/CheckUserExistCommand.cs
using BluffinMuffin.Protocol.Enums;
using Com.Ericmas001.Net.Protocol;

namespace BluffinMuffin.Protocol.Lobby.RegisteredMode
{
    /// <summary>
    /// In the User Creation Process, it's useful to check if the username is already taken. You can then validate the username before actually creating the user. (Username is the name used to connect)
    /// </summary>
    public class CheckUserExistCommand : AbstractLobbyCommand, ICommandWithResponse<CheckUserExistResponse>
    {
        /// <summary>
        /// The display name you want to check the existance of
        /// </summary>
        [ExampleValue("ericmas001")]
        public string Username { get; set; }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public CheckUserExistResponse Response(bool success, BluffinMessageId msgId, string message)
        {
            return new CheckUserExistResponse(this) { Success = success, MessageId = msgId, Message = message };
        }
    }
}

File: Python/bluffin_muffin/protocol/lobby/registered_mode/check_user_exist_command.py
from protocol import AbstractLobbyCommand

__author__ = 'ericmas001@gmail.com'



class CheckUserExistCommand(AbstractLobbyCommand):
    def __init__(self, obj):
        super().__init__(obj)
        self.username = obj['Username']

    def __str__( self ):
        return '{0} ({1})'.format(
            super().__str__(),
            self.username
        )

    def _encode_specific(self, d):
        super()._encode_specific(d)
        d['Username'] = self.username
